nixos/modules/programs/zsh/zsh.nix

```diff
@@ -87,6 +87,15 @@ in
         type = types.bool;
       };
 
+
+      enableGlobalCompInit = mkOption {
+        default = cfg.enableCompletion;
+        description = ''
+          Enable execution of compinit call for all interactive zsh shells.
```

Comment:
Can you add, that this option can be used, if the user wants to extend its `fpath`.

Reply:
Done